### PR TITLE
Add custom quick setting tiles

### LINKED ISSUE #1816: Support tiles in quick settings
### Is your feature request related to a problem? Please describe.
Supporting quick tiles would allow faster control of often-used items.

### Describe the solution you'd like
Here's my proposal how to implement this:
Have a fixed number of tile services. Each service represents a tile. All services are disabled by default.
An activity for managing tiles: A list with entries for each tiles. If a entry is clicked, an ItemPickerActivty is opened and let's the user pick item, action, label and also the tile state (shown/hidden). If the tile state is changed, the corresponding service is enabled. Item, action and label (= tile data) are saved in SharedPrefs.
If the tile is added, the tile service reads the tile data and sets up the tile accordingly.
A click on the tile queues an item action in the existing infrastructure in BackgroundTasksManager.

### Describe alternatives you've considered
Opening a item picker if the tile is added. IMO this is bad UX, especially if the items picker is discarded. The tile could be reset if the tile is removed.

### Additional context
https://developer.android.com/reference/android/service/quicksettings/Tile
https://developer.android.com/reference/android/service/quicksettings/TileService

### PR DESCRIPTION
**Pull request by mueller-ma**
Fixes #1816

TODO:

- [x] Update summaries on tile overview page after changing a tile setting
- [x] Add more icons: ~~Equivalents of https://www.openhab.org/docs/configuration/iconsets/classic/ + some of the icons that are already in the app + a few more (like https://materialdesignicons.com/icon/floor-lamp)~~ I added a few icons from the classic icon set and those that are already used in the app. See https://github.com/openhab/openhab-android/pull/1978/files#diff-fe5fbaf1f49ab56c50596e73323c2b70R159 for missing icons.
- [x] Add more tiles, 8 to 12 in total
- [x] Un-tint openHAB icon in item picker
- [x] Re-enable tiles after reboot

Adding more tiles is just copy and paste, but I wanted to get some feedback before doing that.
There could be more beautiful ways to select an icon than a list with names, but I'm not sure if it's worth the effort to implement one.

### CODE REVIEW

File: mobile/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -481,4 +481,32 @@
...
+    <string name="save">Save</string>
+    <string name="tile_number">Tile %d</string>
+    <string name="tile_error_saving">Please fill out every setting</string>
+    <string name="tile_disabled">Turned off</string>
+    <string name="tile_enable">Turn on</string>
+    <string name="item_item_and_action">Item and action</string>
+    <string name="tile_name">Name</string>
+    <string name="tile_icon">Icon</string>
+    <string name="tile_require_unlock">Require screen to be unlocked</string>
+    <string name="tile_require_unlock_summary_on">If your device is locked, you\'re prompted to unlock it, before the Item will be updated</string>
```
Comment (maniac103) [inline mobile/src/main/res/values/strings.xml:505]: `Prompt for unlocking the device before allowing Item updates`?

File: mobile/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -481,4 +481,32 @@
...
+    <string name="tile_require_unlock_summary_off">The Item may be updated, even if the device is locked</string>
```
Comment (maniac103) [inline mobile/src/main/res/values/strings.xml:506]: `Allow updating Items even if the device is locked`?

File: mobile/src/main/java/org/openhab/habdroid/ui/PreferencesActivity.kt [hunk 3]
```diff
@@ -474,6 +504,18 @@ class PreferencesActivity : AbstractBaseActivity() {
...
         }
 
+        @RequiresApi(Build.VERSION_CODES.N)
+        private fun updateTileSummary() {
+            var activeTileCount = 0
+            for (tileId in 1 until AbstractTileService.TILE_COUNT + 1) {
+                if (prefs.getTileData(tileId) != null) {
+                    activeTileCount++
+                }
+            }
```
[the remark this answers is not in the document]
Comment (mueller-ma) [reply mobile/src/main/java/org/openhab/habdroid/ui/PreferencesActivity.kt]: Picked. With suggestions by AS this became:
````kotlin
val activeTileCount = (1..AbstractTileService.TILE_COUNT)
    .mapNotNull { id -> prefs.getTileData(id) }
    .size
````

File: mobile/src/main/java/org/openhab/habdroid/ui/PreferencesActivity.kt [hunk 4]
```diff
@@ -687,6 +733,179 @@ class PreferencesActivity : AbstractBaseActivity() {
...
     }
 
+
+    @RequiresApi(Build.VERSION_CODES.N)
+    internal class TileOverviewFragment : AbstractSettingsFragment() {
+        override val titleResId: Int @StringRes get() = R.string.tiles_for_quick_settings
+
+        override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
+            addPreferencesFromResource(R.xml.preferences_tile_overview)
+            for (tileId in 1 until AbstractTileService.TILE_COUNT + 1) {
```
[the remark this answers is not in the document]
Comment (mueller-ma) [reply mobile/src/main/java/org/openhab/habdroid/ui/PreferencesActivity.kt]: Done

File: mobile/src/main/res/values/strings.xml (hunk 1 above)
Comment (mueller-ma) [reply mobile/src/main/res/values/strings.xml]: Sounds better, picked

File: mobile/src/main/res/values/strings.xml (hunk 2 above)
Comment (mueller-ma) [reply mobile/src/main/res/values/strings.xml]: Sounds better, picked

File: mobile/src/main/java/org/openhab/habdroid/ui/PreferencesActivity.kt [hunk 5]
```diff
@@ -687,6 +733,179 @@ class PreferencesActivity : AbstractBaseActivity() {
...
+                    val data: TileData? = if (enabledPref.isChecked) {
+                        val itemName = itemAndStatePref.item
+                        val label = itemAndStatePref.label
+                        val state = itemAndStatePref.state
+                        val mappedState = itemAndStatePref.mappedState
+                        val tileLabel = namePref.text
+                        val icon = iconPref.value
+                        val requireUnlock = requireUnlockPref.isChecked
+                        if (itemName.isNullOrEmpty() || state.isNullOrEmpty() || label.isNullOrEmpty() ||
+                            tileLabel.isNullOrEmpty() || mappedState.isNullOrEmpty() || icon.isNullOrEmpty()) {
```
[the remark this answers is not in the document]
Comment (mueller-ma) [reply mobile/src/main/java/org/openhab/habdroid/ui/PreferencesActivity.kt:813]: I tried that, but I cannot see a difference between a disabled or enabled `MenuItem`.

[the remark this answers is not in the document]
Comment (mueller-ma) [reply mobile/src/main/java/org/openhab/habdroid/ui/PreferencesActivity.kt]: https://github.com/openhab/openhab-android/pull/1978/commits/68691b0b9f249a66a8f7f5a4df9b39b451cbe802
IMO the previous solution was better, so I'll probably revert that commit. I just wanted to give you an example to test it.

[the remark this answers is not in the document]
Comment (mueller-ma) [reply mobile/src/main/java/org/openhab/habdroid/ui/PreferencesActivity.kt:839]: Reverted.

**Pull request merged**